### PR TITLE
Enable GPU CI

### PR DESCRIPTION
**Pull request by dhairyagandhi96**
Looking for feedback on this policy for doing GPU CI.

### CODE REVIEW

**Comment by dhairyagandhi96** [comment]
bors try

**Comment by vchuravy** [comment]
Do you need help with enabling this on the Gitlab side?

**Comment by dhairyagandhi96** [comment]
We need to tag the breaking RNN tests. Everything is set up on the GitLab side, thanks a lot @vchuravy! It would be nice if you could take a look at the yml to see if I'm doing anything wrong. Not sure why it timed out. The logs look as expected, minus the `BoundsError` that CuArrays threw up, which I believe is known?

https://gitlab.com/JuliaGPU/Flux.jl/-/jobs/186779653
https://gitlab.com/JuliaGPU/Flux.jl/-/jobs/186779652

also cc. @MikeInnes

**Comment by dhairyagandhi96** [comment]
bors try

**Comment by dhairyagandhi96** [comment]
bors try

**Comment by dhairyagandhi96** [comment]
https://gitlab.com/JuliaGPU/Flux.jl/-/jobs/187336881

File: test/cuda/cuda.jl [hunk 1]
```diff
@@ -41,5 +41,5 @@ end
 if CuArrays.libcudnn != nothing
     @info "Testing Flux/CUDNN"
     include("cudnn.jl")
-    include("curnn.jl")
+    # include("curnn.jl")
```
Comment (MikeInnes) [inline test/cuda/cuda.jl]: Ideally we should still run these locally (and even better if we can run some simple and reliable test on CI). There's probably some gitlab CI environment variable we can look for to run this conditionally.

**Comment by MikeInnes** [comment]
Do we have any idea what's going on with the timeouts, or how to debug?

File: test/cuda/cuda.jl (hunk 1 above)
Comment (dhairyagandhi96) [reply test/cuda/cuda.jl]: Gotcha, I'll make them skip on GitLab. I initially saw the GRU tests come up locally, and so disabled those, but can for sure use some more reliable test on CI

**Comment by dhairyagandhi96** [comment]
The jobs are created and finish well before the timeout but bors can't seem to see them. I'll investigate if there's some config I'm missing.

**Comment by dhairyagandhi96** [comment]
bors ping

**Comment by dhairyagandhi96** [comment]
bors try

**Comment by dhairyagandhi96** [comment]
https://gitlab.com/JuliaGPU/Flux.jl/-/jobs/188190665

**Comment by vchuravy** [comment]
bors try

Thanks to @maleadt for identifying the problem and for Mike to fixing it

**Comment by dhairyagandhi96** [comment]
Bors ping

**Comment by vchuravy** [comment]
bors try

**Comment by dhairyagandhi96** [comment]
Woohoo!

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1,10 +1,37 @@
+before_script:
+  - export CI_CURNN_TEST=true
```
Comment (MikeInnes) [inline .gitlab-ci.yml]: This is a confusing name since it looks like you're enabling the rnn tests here, rather than disabling them.

**Comment by dhairyagandhi96** [comment]
bors try

File: .gitlab-ci.yml (hunk 2 above)
Comment (dhairyagandhi96) [reply .gitlab-ci.yml]: Thanks a lot! I fixed it in the latest commit.

**Comment by dhairyagandhi96** [comment]
Bors r+

**Comment by MikeInnes** [comment]
Cool!

**Pull request merged**